### PR TITLE
Cleanup test dicom instances after tests complete

### PR DESCRIPTION
**Pull request by pengchen0692**
## Description
Cleanup test dicom instances after tests complete 

## Related issues
Addresses [AB#85819](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/85819)

## Testing
Locally tests verify no test instances after test complets

### CODE REVIEW

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs [hunk 1]
```diff
@@ -295,23 +296,15 @@ private async Task<DicomDataset> PostDicomFileAsync(DicomDataset metadataItems =
...
-            // xunit does not seem to call IAsyncDispose.DisposeAsync()
-            // Also wait should be okay in a test context
-            foreach (string studyUid in _createdDicomStudies)
-            {
-                _client.DeleteStudyAsync(studyUid).Wait();
-            }
+        public Task InitializeAsync() => Task.CompletedTask;
 
-            _createdDicomStudies.Clear();
+        public async Task DisposeAsync()
```
Comment (pengchen0692) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs:305]: Simplify to return the async task instead?

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomInstanceId.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Model;
+
+namespace Microsoft.Health.Dicom.Web.Tests.E2E.Common
+{
+    internal class DicomInstanceId
+    {
+        private const StringComparison EqualsStringComparison = StringComparison.Ordinal;
+        public DicomInstanceId(string partitionName, string studyInstanceUid, string seriesInstanceUid, string sopInstanceUid)
```
Comment (pengchen0692) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomInstanceId.cs:17]: move `partitionName` to last for consistency

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomInstancesManager.cs [hunk 3]
```diff
@@ -4,26 +4,28 @@
...
-using Microsoft.Health.Dicom.Core.Features.Model;
 
 namespace Microsoft.Health.Dicom.Web.Tests.E2E.Common
 {
     internal class DicomInstancesManager : IAsyncDisposable
     {
+
         private readonly IDicomWebClient _dicomWebClient;
-        private readonly HashSet<InstanceIdentifier> _instanceIds;
+        private readonly ConcurrentBag<DicomInstanceId> _instanceIds;
```
Comment (pengchen0692) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomInstancesManager.cs:23]: `ConcurrentBag`  to `IList`

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs (hunk 1 above)
Comment (pengchen0692) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/QueryTransactionTests.cs:305]: not possible. _instancesManager.DisposeAsync() is ValueTask

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomInstancesManager.cs (hunk 3 above)
Comment (pengchen0692) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomInstancesManager.cs:23]: impossible

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/StoreTransactionTests.cs [hunk 4]
```diff
@@ -135,7 +138,7 @@ await using (MemoryStream stream = _recyclableMemoryStreamManager.GetStream())
                     multiContent.Add(validByteContent);
                 }
 
-                using DicomWebResponse<DicomDataset> response = await _client.StoreAsync(multiContent);
+                using DicomWebResponse<DicomDataset> response = await _instancesManager.StoreAsync(multiContent, instanceId: DicomInstanceId.FromDicomFile(validFile));
```
Comment (alishahahmed) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/StoreTransactionTests.cs:141]: Why is providing instanceId necessary here?

Comment (pengchen0692) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/StoreTransactionTests.cs:141]: `DicomInstancesManager` need the id to know which dicom file to delete, unfortunately we couldn't infer it from input, so need additional parameter for it.

Comment (alishahahmed) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/StoreTransactionTests.cs:141]: Makes sense, thanks!

**Review by alishahahmed — APPROVED** [review]

**Pull request merged**